File: rl/engine/strategy.py
from typing import List, TYPE_CHECKING

if TYPE_CHECKING:
    from engine import Card, Trick, Suit


class Strategy:
    def choose_card(
            self, hand: List['Card'], game_history: List['Trick'], trick: 'Trick', trump_suit: 'Suit'
    ) -> int:
        raise NotImplementedError

File: engine/src/main/java/com/mjaseem/trick/strategy/Strategy.java
package com.mjaseem.trick.strategy;

import com.mjaseem.trick.engine.Card;
import com.mjaseem.trick.engine.GameHistory;
import com.mjaseem.trick.engine.Records;
import com.mjaseem.trick.engine.Trick;

import java.util.List;

// Strategy interface
public interface Strategy {

    int chooseCard(List<Card> hand, GameHistory history, Trick trick, Records.Suit trumpSuit);
}
